util/make_new_dif/dif_template.h.tpl

```diff
@@ -0,0 +1,209 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+<%doc>
+    This templates requires the following Python objects to be passed:
+
+    1. ip: See util/make_new_dif.py for the definition of the `ip` obj.
+</%doc>
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_${ip.name_upper}_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_${ip.name_upper}_H_
+
+// This file is the "DIF Library header template", which provides a base for
+// building a DIF for a new peripheral, defining all of the declarations that
+// would be expected of a DIF library as described in the README.md.
+//
+// This file should be instantiated with the `util/make_new_dif.py` script.
+//
+// The script also includes additional options for controlling how the template
+// is instantiated. After the script runs, delete this comment, the #error
+// below, and any unnecessary definitions. Remember to run clang-format!
+//
+// Note that this file does not parse as C.
+
+#error "This file is a template, and not real code."
+
+/**
+ * @file
+ * @brief <a href="/hw/ip/${ip.name_snake}/doc/">${ip.name_long_upper}</a> Device Interface Functions
+ */
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * A toggle state: enabled, or disabled.
+ *
+ * This enum may be used instead of a `bool` when describing an enabled/disabled
+ * state.
+ */
+typedef enum dif_${ip.name_snake}_toggle {
+  /*
+   * The "enabled" state.
+   */
+  kDif${ip.name_camel}ToggleEnabled,
+  /**
+   * The "disabled" state.
+   */
+  kDif${ip.name_camel}ToggleDisabled,
+} dif_${ip.name_snake}_toggle_t;
```

Comment:
Is it on your list to factor out `toggle_t` into a shared header...?

Reply:
Yes thats is the plan, and perhaps some other items as well (but in a different future PR). For now, moving this to the autogenerated include file (dif_<ip>_autogen.inc.tpl), since no need to be in the one that is semi-auto-generated & manually-tweaked.